client/trino-cli/src/main/java/io/trino/cli/Console.java

```diff
@@ -104,7 +108,16 @@ public Integer call()
 
     public boolean run()
     {
-        ClientSession session = clientOptions.toClientSession();
+        // map option names to URL params and restrict those set explicitly by the user
+        CommandLine.ParseResult pr = spec.commandLine().getParseResult();
+        List<CommandLine.Model.OptionSpec> options = spec.options();
+        List<String> providedOptions = options.stream()
+                .filter(pr::hasMatchedOption)
+                .map(CommandLine.Model.OptionSpec::longestName)
+                .collect(Collectors.toList());
+
+        TrinoUri uri = clientOptions.getTrinoUri(providedOptions);
```

Comment:
This has a nice side effect of always converting all CLI options into URLs that _can_ be used as JDBC URLs. Would it be useful to log the resulting connection URL on a debug level?

Reply:
If it won't expose the password I'm for it.